icu4j/main/classes/core/src/com/ibm/icu/util/ULocale.java

```diff
@@ -507,6 +507,16 @@ public static ULocale createCanonical(String nonCanonicalID) {
         return new ULocale(canonicalize(nonCanonicalID), (Locale)null);
```

Comment:
Please also update the API docs of the existing C++ and Java *canonical* functions to say "... according to CLDR" or whatever wording we end up with on the new functions.